Patch-Source: https://github.com/gdnsd/gdnsd/commit/4df9ddfedfb186137a88e154497f619306b8ddb7
--
From 4df9ddfedfb186137a88e154497f619306b8ddb7 Mon Sep 17 00:00:00 2001
From: Brandon Black <bblack@wikimedia.org>
Date: Mon, 13 Jan 2025 13:02:43 -0600
Subject: [PATCH] Fix Net::DNS 1.49 testsuite breakage

As of 1.49, apparently the previously readonly call
RR::OPT->option('foo') will, as a side effect, *create* the option
"foo" with an option-length of zero if it doesn't already exist.

One part of our testsuite driver used this call to check for the
existence of the "cookie" option while parsing packets for tests
which did not use the "cookie" option, resulting in failures from
the artificially-created "cookie" option.  This commit switches
the option() calls to use direct access to the option hash
instead, avoiding the side-effect.
---
 t/_GDT.pm | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/t/_GDT.pm b/t/_GDT.pm
index f9c2f2f6..be9c84bb 100644
--- a/t/_GDT.pm
+++ b/t/_GDT.pm
@@ -833,11 +833,12 @@ sub _compare_rrsets {
     } elsif($rrtype eq 'CNAME') {
         $limit = 1;
     } elsif($rrtype eq 'OPT') {
+	my $cookie_num = ednsoptionbyname('COOKIE');
         $_lastacookie = undef;
-        my $ccookieval = $c_rrset->[0]->option('COOKIE');
+        my $ccookieval = $c_rrset->[0]->{option}{$cookie_num};
         if (defined $ccookieval) {
             my $ccookielen = length($ccookieval);
-            my $acookieval = $a_rrset->[0]->option('COOKIE');
+            my $acookieval = $a_rrset->[0]->{option}{$cookie_num};
             if (defined $acookieval) {
                 my $acookielen = length($acookieval);
                 return "Bad server cookie len ($acookielen != $ccookielen)" if ($acookielen != $ccookielen);
